control-plane/api-gateway/binding/route_binding.go

```diff
@@ -457,7 +457,7 @@ func consulCondition(generation int64, status api.ConfigEntryStatus) *metav1.Con
 	for _, c := range status.Conditions {
 		// we only care about the top-level status that isn't in reference
 		// to a resource.
-		if c.Type == "Accepted" && c.Resource.Name == "" {
+		if c.Type == "Accepted" && c.Resource == nil { //c.Resource.Name == "" {
```

Comment:
Need to review this

Reply:
I think this accomplishes the same result, we are still checking the highest level status, and it looks like that would not have a resource (because the resource is the gateway)